hermes_core/util/cdf.py

```diff
@@ -0,0 +1,612 @@
+"""
+This module provides CDF file functionality.
+
+"""
+import datetime
+from pathlib import Path
+import yaml
+from spacepy import pycdf
+import numpy as np
+import hermes_core
+from hermes_core import log
+from hermes_core.util import util
+from hermes_core.util.exceptions import warn_user
+
+DEFAULT_SCHEMA_FILE = "hermes_default_schema.yaml"
+DEFAULT_ATTRIBUTES_FILE = "hermes_default_attributes.yaml"
+
+
+class CDFWriter:
+    """
+    Python Class to Convert abstract Data Structures to CDF file
+    formats compliant with SpacePy's pyCDF Module. These CDF Files
+    can then be indexed and shared through SPDF.
+
+    The purpose of this class is to provide an intermediate interafce to generate CDF Files.
+    One limitation of the pyCDF library is that you can not create CDF-like data structures
+    in memory without generating and saving a local `.cdf` File. This class provides an
+    interface to manipulate CDF-like data structures in memory before the generation and saving
+    of a local `.cdf` file.
+
+    This class uses an internal native python `dict` data structure to provide a CDF-like
+    data structure that can be manipulated in memory. The internal `dict` used `self.global_attrs`
+    is formatted in the following way.
+
+    ```py
+    self.global_attrs = {
+        'global_attribute_name1': 'global_attribute_value1',
+        'global_attribute_name2': 'global_attribute_value2',
+    }
+
+    ```
+    """
+
+    def __init__(self):
+
+        # Current Working CDF File
+        self.target_cdf_file = None
```

Comment:
why do you need this? Doesn't the file name get generated automatically?

Reply:
The goal here, and similarly for splitting the `to_cdf` and `save_cdf` functions is to give users the chance to manipulate the `pycdf` data structure directly before closing the file. 

If you don't see this as a use case, then I can remove this member, and combine both functions into a single `write_cdf`. Let me know if you see this as a use case.